api/nakadi-event-bus-api.yaml

```diff
@@ -1503,11 +1541,37 @@ definitions:
         type: string
         default: 'random'
 
+      compatibility_mode:
+        description: |
+          Compatibility modes provide a mean for event owners to evolve their schema, given changes
+          respect the semantics defined by this field.
+
+          It's designed to be flexible enough so that producers can evolve their schemas while not
+          inadvertently breaking existent consumers.
+
+          Once defined, the compatibility mode is fixed, since otherwise it would break a predefined contract,
+          declared by the producer.
+
+          List of available compatibility modes:
+
+          - 'deprecated': Schema changes are not allowed. This is used only by legacy event types. This option is not
+            available for new event types. Schema validation is lenient and allow fields that are not defined. Also,
+            it's possible to use the full json schema specification for defining schemas.
+
+          - 'none': Any schema modification is accepted, even if it might break existing producers or consumers.
+
+          - 'compatible': Schema changes never break existing consumers. Every event published since the first version
+            is still valid based on the newest schema. When in compatible mode, it's allowed to add new optional
+            properties and definitions to an existing schema, but no other changes are allowed. If additionalProperties
```

Comment:
@rcillo Thanks -- but I am somehow worried about the array based approach as well: 

1. It shares the structural problem of `additionalProperties =true` --- new attributes can sneak in without being explicitly defined
2. We have even less control over data types (all values have to be strings)
3. It's harder to read and process.

Why couldn't the schema for your example be as follows?

`{
  "type": "object",
  "properties": {
    "sku": {
      "type": "string"
    },
    "product_attributes": {
      "type": "object",
      "properties": {
        "color": {
          "type": "string"
        },
        "size": {
          "type": "string"
        }
      },
      "additionalProperties": false
    }
  },
  "additionalProperties": false
}`